### PR TITLE
Solve fragmentation issue with remesh_along_isoline

### PR DESCRIPTION
**Pull request by n-taka**
[Describe your changes and what you've already done to test it.]
Previous remesh_along_isoline function causes fragmentation issue (the resulting mesh gets disconnected).
This commit solves it.

#### Check all that apply (change to `[x]`)
- [x] All changes meet [libigl style-guidelines](https://libigl.github.io/style-guidelines/).
- [x] Adds new .cpp file.
- [x] Adds corresponding unit test.
- [ ] Adds corresponding python binding.
- [ ] This is a minor change.

### CODE REVIEW

**Comment by jdumas** [comment]
Do you have a test-case where the problem happens? If so, could you also add this as a unit test for this function?

File: include/igl/remesh_along_isoline.cpp [hunk 1]
```diff
@@ -71,6 +71,7 @@ template <
   }
 
   // Loop over each face
+  std::unordered_map<int, std::unordered_map<int, int>> edgeToBirthVert;
```
Comment (BruegelN) [inline include/igl/remesh_along_isoline.cpp]: Even though unordered_map is a C++11 feature you need to `#include <unordered_map>` especially on gcc 4.8 which is used by default in Ubuntu 14.04 (travis-ci).

Comment (n-taka) [reply include/igl/remesh_along_isoline.cpp]: Thank you! I added #include<unordered_map> line.

**Comment by n-taka** [comment]
Hi, the previous implementation causes fragmentation issue almost all time remesh happens.
To make it clear, I added simple unit test for remesh_along_isoline, which checks that the number of face connected components is unchanged.

**Comment by BruegelN** [comment]
I wonder if there are any news on this PR? It's a bit impractical if a PR contains many different things. To play around with it I've cherry-picked the relevant commits into a branch of mine: https://github.com/BruegelN/libigl/tree/fix-remesh_along_isoline. There are case where the current implementation for `igl::remesh_along_isoline()` causes fragmentation and this PR resolves it.

**Comment by jdumas** [comment]
I believe it would help if we were to add such test cases as a unit test.

**Comment by BruegelN** [comment]
Hi @jdumas I'll have a look at this during the weekend.
Maybe @n-taka wasn't aware that his branch is not yet merged?!

**Comment by jdumas** [comment]
Superseded by #1440

**Pull request closed without merging**